File: app/src/main/java/com/example/cobe/cryptonews/api/ArticlesInteractorInterface.java
package com.example.cobe.cryptonews.api;


import com.example.cobe.cryptonews.model.Article;

import java.util.List;

/**
 * Created by cobe on 06/04/2018.
 */

public interface ArticlesInteractorInterface {

    interface ResponseInterface {

        void onArticlesSuccess(List<Article> articles);

        void onArticlesError();
    }

    void getArticles(ResponseInterface responseInterface);

    void getSearchedArticles(ResponseInterface responseInterface, String text, String date);

}

File: app/src/main/java/com/example/cobe/cryptonews/interaction/ArticlesInteractorInterface.kt
package com.example.cobe.cryptonews.interaction

import com.example.cobe.cryptonews.model.Article

/**
 * Created by cobe on 11/04/2018.
 */
interface ArticlesInteractorInterface {

    interface ResponseInterface<in T> {

        fun onArticlesSuccess(data: T)

        fun onArticlesError()
    }

    fun getArticles(responseInterface: ResponseInterface<List<Article>>)

    fun getSearchedArticles(responseInterface: ResponseInterface<List<Article>>, text: String, date: String)

}